### PR TITLE
feat: :sparkles: show a notification item in the settings page

### PR DESCRIPTION
**Pull request by matteoscurati**
## **Description**

This PR introduces the `Notifications` option within the `settings page`. Clicking on this option will render the user to the `notifications/settings` page.
Specifically:
1. If basic functionalities are not active, the option disappears from the menu
2. If notifications are not active, the user will still be redirected to the `notifications/settings page`, where they can enable them
3. If notifications are active, the user will access the `notifications/settings` page, where they can select their options

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/26843?quickstart=1)

## **Related issues**

N/A

## **Manual testing steps**

1. Go to the settings page.
2. Find the `Notifications` option
3. Verify that the `Notifications` option links to the `notifications/settings` page
4. Verify that the `Notifications` option is hidden if basic functionalities are off

### **Before**

Screenshot 2024-10-23 at 16 26 12

### **After**

Screenshot 2024-10-23 at 16 41 30

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ui/pages/notifications-settings/notifications-settings.tsx [hunk 1]
```diff
@@ -82,7 +81,7 @@ export default function NotificationsSettings() {
             ariaLabel="Back"
             iconName={IconName.ArrowLeft}
             size={ButtonIconSize.Sm}
-            onClick={() => history.push(NOTIFICATIONS_ROUTE)}
+            onClick={() => history.go(-1)}
```
Comment (Prithpal-Sooriya) [inline ui/pages/notifications-settings/notifications-settings.tsx:84]: I don't see this used elsewhere in extension.
Can we use `history.back()`?

Comment (matteoscurati) [reply ui/pages/notifications-settings/notifications-settings.tsx]: it seems we should use `history.goBack()` 🤷

**Review by mathieuartu — DISMISSED** [review]
LGTM!

**Review by Prithpal-Sooriya — DISMISSED** [review]
Can we update the PR description to include before/after image or a recording (just to help testing purposes)

File: ui/pages/settings/settings.component.js [hunk 2]
```diff
@@ -326,6 +328,14 @@ class SettingsPage extends PureComponent {
       },
     ];
 
+    if (useExternalServices) {
```
[the remark this answers is not in the document]
Comment (matteoscurati) [reply ui/pages/settings/settings.component.js:331]: Yep

File: ui/pages/notifications-settings/notifications-settings.tsx (hunk 1 above)
Comment (Prithpal-Sooriya) [reply ui/pages/notifications-settings/notifications-settings.tsx]: Aha nice find!

File: ui/pages/notifications-settings/notifications-settings.tsx [hunk 3]
```diff
@@ -82,7 +81,7 @@ export default function NotificationsSettings() {
             ariaLabel="Back"
             iconName={IconName.ArrowLeft}
             size={ButtonIconSize.Sm}
-            onClick={() => history.push(NOTIFICATIONS_ROUTE)}
+            onClick={() => history.goBack()}
```
[the remark this answers is not in the document]
Comment (matteoscurati) [reply ui/pages/notifications-settings/notifications-settings.tsx]: @Prithpal-Sooriya good catch! Look at the new implementation :)

**Review by Prithpal-Sooriya — APPROVED** [review]

**Review by mathieuartu — APPROVED** [review]

**Pull request merged**